### DISCUSSION TITLE
Pomodoro Timer for Each Task

### DISCUSSION THREAD

**Discussion by Spikeysanju** [Spikeysanju/Einsen#86] [Ideas]
Checkout this issue for more info  https://github.com/Spikeysanju/Einsen/issues/79

**Comment by SachinKasaraddi** [comment]
## Why i like to contribute
* I want to start contributing to open source projects.
* Learn latest technologies while contributing to this awesome project. :book:
* Pomodoro technique is close to my heart topic, as i follow this [technique personally as well.](https://sachinkasaraddi.com/posts/debounce-and-navigatesafe-together/) :smile: 
*  Engage with android community outside the office world. :tada:  

## How i would like to contribute
* Be clear and be on same page with @Spikeysanju, on what we want to achieve before starting the work.
* Technical details discussion on high level (If any).
* Dedicate (1-2hrs / day) :alarm_clock:
* Open for suggestions (if any)

## What i like about this feature

* Instead of providing the list with prioritised tasks won't make user productive enough. :scroll:
* Providing user the flexibility and focus on each task with regular intervals, instead of going down the **rabit hole** (from one task to another). 
* :heart: feature to contribute.

## QUESTIONS
-   The idea behind pomodoro technique is to focus on one single task at a time,  instead of we deciding on behalf of user (task with upcoming due date having option to start pomodoro), why can’t user start his prioritised task himself :thinking:?
	-   (Suggestion: : Hence can we move this pomodoro card to task detail page instead. )
	-  Still user will be able to start pomodoro in **ALL TASK** screen with :alarm_clock: icon on each task. 
* Could you elaborate more on  `Show overall time` to complete all task.
* Can user change the interval time from 25 to less or more?
* Can user pause the timer, if he has some disturbance or emergency to not continue on task?
* Will timer be running irrespective of app being closed? (Foreground service).
* Pomodoro complete assumes the task is complete?
	* If yes can we provide the confirm dialog to check whether task is completed?

> Note: Open to changes on How to work or any guidelines to follow :pray:.

**Reply by Spikeysanju** [reply]
@SachinKasaraddi  First of thanks for your interest. It really makes me happy to interact with folks like you 😄 . Since this feature is going to give so much value to the people we need to be clear with our requirements and technical details before starting this.

> Instead of providing the list with prioritised tasks won't make user productive enough. 📜 
Yes I 💯 agreed to this point. But this application was designed &  built to mix both `Prioritisation` + `Pomodoro timer` in mind.  If you see we have successfully completed the Task prioritisation so user can easily table to focus on what to do next.

With `Pomodoro timer ⏰`  feature it will give user an extra ability to `getting things done 🔥` . So this is going to be a game changer for this app. 

### Answers for your questions 👇 

1. I agreed 💯 . So by default we will show `Pomodoro timer ⏰` for the `1st task` only (Because it the task with upcoming due date). But still user can able to drag any task item card to top. 

- Now that task item will have `Pomodoro timer ⏰`. I will share the `prototype design` for better understanding. There's still a room for improvement. Feel free share your thoughts on this 😉
- Yes we need to show the `Pomodoro timer ⏰` icon on both `All Task Screen` & `Task Details Screen`. I will update the designs. Thanks for the suggestion 🙌 

2. Show overall time - This is basically the total amount of time will take to complete the task. So it's the sum of All Task Item time. 

- For ex - By default we're going to add `25 mins` for each task. Now when `clicks` on the `timer text` from `All Task Screen` we will show a `Timer dialog` or `Slider` to `increase or decrease` the timer. 

- But for now we can `skip this feature` by adding another field in `add task screen`. We will go-ahead with `25 mins` for each if user didn't add any value when `creating a task`.

3. Yes we need to provide an option to `pause`, `play` & `+ 5 mins` or `-5 mins` etc.

4. Yes we need to use Foreground services to run the timer even if the app was closed.

5. We never know how long each task will take. Sometimes users `forgot` to update the timer so it won't good to assume that user will complete the task `within this time frame`. So I need to check how other peeps are handling this scenario. As you said we can confirm this by showing a `confirm dialog 👍` .

- Again thanks for bringing this up. These question are really interesting & makes me think about handling some edge cases. I will update the designs and update it here 🙌 . Since this task is really `big` & `interesting`. 

- What I would suggest you is we can do a collab by `splitting up` the task. I will break the task into `smaller chunks` so that we can pickup it 👍

**Reply by SachinKasaraddi** [reply]
@Spikeysanju Yeah sure we can `split task` and work on smaller chunks. Would be waiting for the first small task to contribute 👍

**Reply by SachinKasaraddi** [reply]
@Spikeysanju Any update on this task to start recently?

**Reply by Spikeysanju** [reply]
@SachinKasaraddi Thanks for your interest. I'm currently working on the Einsen app widgets. Once it is done. I will be updating you here 🙌
